R/dash.R

```diff
@@ -573,32 +627,178 @@ Dash <- R6::R6Class(
     # convenient fiery wrappers
     # ------------------------------------------------------------------------
     run_server = function(host = Sys.getenv('DASH_HOST', "127.0.0.1"),
-                          port = Sys.getenv('DASH_PORT', 8050),
+                          port = Sys.getenv('DASH_PORT'),
                           block = TRUE,
                           showcase = FALSE,
+                          use_viewer = FALSE,
                           dev_tools_prune_errors = TRUE,
-                          debug = FALSE,
-                          dev_tools_ui = NULL,
-                          dev_tools_props_check = NULL,
+                          debug = Sys.getenv('DASH_DEBUG'),
+                          dev_tools_ui = Sys.getenv('DASH_UI'),
+                          dev_tools_props_check = Sys.getenv('DASH_PROPS_CHECK'),
+                          dev_tools_hot_reload = Sys.getenv('DASH_HOT_RELOAD'),
+                          dev_tools_hot_reload_interval = Sys.getenv('DASH_HOT_RELOAD_INTERVAL'),
+                          dev_tools_hot_reload_watch_interval = Sys.getenv('DASH_HOT_RELOAD_WATCH_INTERVAL)'),
+                          dev_tools_hot_reload_max_retry = Sys.getenv('DASH_HOT_RELOAD_MAX_RETRY'),
+                          dev_tools_silence_routes_logging = NULL,
                           ...) {
-      self$server$host <- host
-      self$server$port <- as.numeric(port)
+      if (exists("dash_nested_fiery_server", env=parent.frame(1))) {
+        # fiery is attempting to launch a server within a server, abort gracefully
+        return(NULL)
+      }
 
-      if (is.null(dev_tools_ui) && debug || isTRUE(dev_tools_ui)) {
-        self$config$ui <- TRUE
+      getServerParam <- function(value, type, default) {
+        if (!is.null(value) && toupper(value) %in% c("TRUE", "FALSE"))
```

Comment:
```suggestion
        if (type == "logical" && !is.null(value) && toupper(value) %in% c("TRUE", "FALSE"))
```
Maybe also add a clause
```
if (type == "numeric")
    value <- as.numeric(value)
```
for the hot reload intervals; `as.integer` probably has to stay where it is since that doesn't appear to map onto a type...
```

Reply:
R is interesting in that there is no "type" function, but you can verify that an integer is of mode `numeric` and storage mode `integer`:

```
> foo <- 1L
> mode(foo)
[1] "numeric"
> storage.mode(foo)
[1] "integer"
```

The `col.info` function I wrote a while back attempts to present a summary with all the type/mode information I generally want for vectors and data.frames:

https://github.com/rpkyle/cscmisc/blob/master/R/col_info.R

Reply:
fixed in ba7c01589f3cdd5d084a3a6b7f1ac733cb4d6458